src/Data/ArrayBuffer/Typed.purs

```diff
@@ -1,87 +1,544 @@
+-- |   `Data.ArrayBuffer.ArrayBuffer.slice`.
+-- | - `empty` and `fromArray` are methods for creating pure typed arrays
+-- |
+-- | #### Modification
+-- |
+-- | - `fill`, `set`, and `setTyped` are methods for assigning values from external sources
+-- | - `map` and `traverse` allow you to create a new array from the existing values in another
+-- | - `copyWithin` allows you to set values to the array that exist in other parts of the array
+-- | - `filter` creates a new array without the values that don't pass a predicate
+-- | - `reverse` modifies an existing array in-place, with all values reversed
+-- | - `sort` modifies an existing array in-place, with all values sorted
+-- |
+-- | #### Access
+-- |
+-- | - `elem`, `all`, and `any` are functions for testing the contents of an array
+-- | - `unsafeAt`, `hasIndex`, and `at` are used to get values from an array, with an offset
+-- | - `foldr`, `foldrM`, `foldr1`, `foldr1M`, `foldl`, `foldlM`, `foldl1`, `foldl1M` all can reduce an array
+-- | - `find` and `findIndex` are searching functions via a predicate
+-- | - `indexOf` and `lastIndexOf` are searching functions via equality
+-- | - `slice` returns a new typed array on the same array buffer content as the input
+-- | - `subArray` returns a new typed array with a separate array buffer
+-- | - `toString` prints to a CSV, `toString'` allows you to supply the delimiter
+-- | - `toArray` returns an array of numeric values
+class BinaryValue a t <= TypedArray (a :: ArrayViewType) (t :: Type) | a -> t where
+  -- | View mapping the whole `ArrayBuffer`.
+  whole :: ArrayBuffer -> ArrayView a
+  -- | View mapping the rest of an `ArrayBuffer` after an index.
+  remainder :: ArrayBuffer -> ByteOffset -> Effect (ArrayView a)
+  -- | View mapping a region of the `ArrayBuffer`.
+  part :: ArrayBuffer -> ByteOffset -> Length -> Effect (ArrayView a)
+  -- | Creates an empty typed array, where each value is assigned 0
+  empty :: Length -> ArrayView a
+  -- | Creates a typed array from an input array of values, to be binary serialized
+  fromArray :: Array t -> ArrayView a
+  -- | Fill the array with a value
+  fill :: ArrayView a -> t -> Range -> Effect Unit
+  -- | Stores multiple values into the typed array
+  set :: ArrayView a -> Maybe Offset -> Array t -> Effect Unit
+  -- | Maps a new value over the typed array, creating a new buffer and typed array as well.
+  map :: (t -> Offset -> t) -> ArrayView a -> ArrayView a
+  -- | Traverses over each value, returning a new one
+  traverse :: (t -> Offset -> Effect t) -> ArrayView a -> Effect (ArrayView a)
+  -- | Traverses over each value
+  traverse_ :: (t -> Offset -> Effect Unit) -> ArrayView a -> Effect Unit
+  -- | Test a predicate to pass on all values
+  all :: (t -> Offset -> Effect Boolean) -> ArrayView a -> Effect Boolean
+  -- | Test a predicate to pass on any value
+  any :: (t -> Offset -> Effect Boolean) -> ArrayView a -> Effect Boolean
```

Comment:
Why are the callbacks effectful here?

Reply:
I was just trying to stay true to how the foreign js would operate; but you're totally right - effectful behaviour like this should be left to a traverseWithIndex.